Mathlib/NumberTheory/ModularForms/JacobiTheta/Bounds.lean

```diff
@@ -249,11 +249,7 @@ lemma isBigO_atTop_F_int_zero_sub (a : UnitAddCircle) : ∃ p, 0 < p ∧
   obtain ⟨a, ha, rfl⟩ := a.eq_coe_Ico
   obtain ⟨p, hp, hp'⟩ := isBigO_atTop_F_nat_zero_sub ha.1
   obtain ⟨q, hq, hq'⟩ := isBigO_atTop_F_nat_zero_sub (sub_nonneg.mpr ha.2.le)
-  have ha' : (a : UnitAddCircle) = 0 ↔ a = 0 := by
-    rw [← AddCircle.coe_eq_coe_iff_of_mem_Ico (hp := ⟨zero_lt_one' ℝ⟩), QuotientAddGroup.mk_zero]
-    · rw [zero_add]; exact ha
-    · simp
-  simp_rw [ha']
+  simp_rw [AddCircle.coe_eq_zero_iff_of_mem_Ico ha]
```

Comment:
Here the type of `ha` is `a ∈ Ico 0 1`.
The signature of [`AddCircle.coe_eq_zero_iff_of_mem_Ico`](https://github.com/leanprover-community/mathlib4/blob/75368bee46034cbb2e556461d176e4a41391cc9e/Mathlib/Topology/Instances/AddCircle.lean#L217) is
```lean
AddCircle.coe_eq_zero_iff_of_mem_Ico.{u_1} {𝕜 : Type u_1} [LinearOrderedAddCommGroup 𝕜] {p : 𝕜} [hp : Fact (0 < p)]
  {a : 𝕜} [Archimedean 𝕜] (ha : a ∈ Ico 0 p) : ↑a = 0 ↔ a = 0

```

That is to say: this change does not introduce any defeq abuse.